plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -100,6 +100,12 @@ public function create( &$order ) {
 			$order->set_order_key( wc_generate_order_key() );
 		}
 		parent::create( $order );
+
+		// Do not fire 'woocommerce_new_order' for draft statuses.
+		if ( in_array( $order->get_status( 'edit' ), array( 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
```

Comment:
should we extract this list into a common method (like we have for paid statuses `wc_get_is_paid_statuses` )?

Reply:
> It's not executed when a new order is created with draft status.
> It is executed when it's moved to a non-draft status.

I created a new issue just for this: https://github.com/woocommerce/woocommerce/issues/48080. This PR was done to prevent this from triggering when checkout is visited and to consolidate behavior between CPT and HPOS.